pkg/modules/application_mobility.go

```diff
@@ -512,68 +514,67 @@ func AppMobilityVelero(ctx context.Context, isDeleting bool, op utils.OperatorCo
 					if c.UseSnapshot {
 						useSnap = true
 					}
-					if c.CleanUpCRDs {
-						cleanUp = true
-					}
 					if c.DeployNodeAgent {
 						nodeAgent = true
 					}
 					for _, env := range c.Envs {
 						if strings.Contains(AppMobObjStoreSecretName, env.Name) {
-							credName = env.Value
+							envCredName = env.Value
 						}
 					}
 					for _, cred := range c.ComponentCred {
 						if cred.Enabled {
-							credName = string(cred.Name)
+							compCredName = string(cred.Name)
+							foundCred, er := utils.GetSecret(ctx, compCredName, cr.Namespace, ctrlClient)
+							if foundCred == nil {
+								err := CreateVeleroAccess(ctx, isDeleting, op, cr, ctrlClient)
+								if err != nil {
+									return fmt.Errorf("\n Unable to deploy velero-secret for Application Mobility: %v\n", err)
+								}
+							} else if foundCred != nil {
+									log.Errorw("\n The secret : ", foundCred.Name, " already exists in the provided namespace")
+									log.Errorw("\n Rename the Secret : ", foundCred.Name, " or if you want to re-use it, plug the secret name in existing-cred argument in samples file")
```

Comment:
I think if we rename credentials.enabled, this message could be more helpful. Current issue is that credentials.enabled will override existing_creds, so if customer follows this message's advice, they will still hit the error. We should mention both fields that need to be adjusted in error message

Reply:
Customer wont hit this error in that case. This error will only show when they try to create credential and it already exists. if they set credentials.enabled to False, its gonna skip this else if statement.

Reply:
True, but this error message doesn't mention setting credentials.enabled to False. 
```Rename the Secret : ", foundCred.Name, " or if you want to re-use it, plug the secret name in existing-cred argument in samples file"```
If customer hits this error, they'll update existing-cred argument, but will hit issue again because they didn't change credentials.enabled